with_recoverer.go

```diff
@@ -0,0 +1,53 @@
+package runnable
+
+import (
+	"context"
+	"fmt"
+)
+
+type RecoveryReporter interface {
+	Report(ctx context.Context, rec interface{})
+}
+
+// NoopReporter
+// Used to continue running go routine  and do nothing
+type NoopReporter struct{}
+
+func (*NoopReporter) Report(ctx context.Context, rec interface{}) {}
+
+type recoverer struct {
+	reporter RecoveryReporter
+}
+
+func WithRecoverer(reporter RecoveryReporter) Option {
+	return &recoverer{
+		reporter: reporter,
+	}
+}
+
+func (rec *recoverer) apply(r *runnable) {
+	runFunc := r.runFunc
```

Comment:
Maybe more straightforward?
```suggestion
	originalRunFunc := r.runFunc
```